sklearn/linear_model/tests/test_huber.py

```diff
@@ -0,0 +1,180 @@
+from sklearn.utils.testing import assert_greater
+
+from sklearn.datasets import make_regression
+from sklearn.linear_model import HuberRegressor, Ridge, SGDRegressor
+from sklearn.linear_model.huber import _huber_loss_and_gradient
+
+rng = np.random.RandomState(0)
+
+
+def make_regression_with_outliers(n_samples=50, n_features=20):
+    X, y = make_regression(
+        n_samples=n_samples, n_features=n_features,
+        random_state=0, noise=10.0)
+
+    # Replace 10% of the sample with noise.
+    num_noise = int(0.1 * n_samples)
+    random_samples = rng.randint(0, n_samples, num_noise)
+    X[random_samples, :] = rng.normal(0, 1, (num_noise, X.shape[1]))
+    return X, y
+
+
+def test_huber_equals_ridge_for_high_epsilon():
+    """Test that Ridge matches HuberRegressor for large epsilon"""
+    X, y = make_regression_with_outliers()
+    ridge = Ridge(fit_intercept=True, alpha=0.0, solver='svd')
+    ridge.fit(X, y)
+    huber = HuberRegressor(fit_intercept=True, epsilon=1e2, alpha=0.0)
+    huber.fit(X, y)
+    assert_almost_equal(huber.coef_, ridge.coef_, 2)
+    assert_almost_equal(huber.intercept_, ridge.intercept_, 2)
+
+
+def test_huber_gradient():
+    """Test that the gradient calculated by _huber_loss_and_gradient is ok"""
+
+    X, y = make_regression_with_outliers()
+    loss_func = lambda x, *args: _huber_loss_and_gradient(x, *args)[0]
+    grad_func = lambda x, *args: _huber_loss_and_gradient(x, *args)[1]
+
+    # Check using optimize.check_grad that the gradients are equal.
+    for _ in range(5):
+        # Check for both fit_intercept and otherwise.
+        for n_features in [X.shape[1] + 1, X.shape[1] + 2]:
+            w = rng.randn(n_features)
+            w[-1] = np.abs(w[-1])
+            grad_same = optimize.check_grad(
+                loss_func, grad_func, w, X, y,  0.01, 0.1)
+            assert_almost_equal(grad_same, 1e-6, 4)
+
+
+def test_huber_sample_weights():
+    """Test sample_weights implementation in HuberRegressor"""
+
+    X, y = make_regression_with_outliers()
+    huber = HuberRegressor(fit_intercept=True, alpha=0.1)
+    huber.fit(X, y)
+    huber_coef = huber.coef_
+    huber_intercept = huber.intercept_
+
+    huber.fit(X, y, sample_weight=np.ones(y.shape[0]))
+    assert_array_almost_equal(huber.coef_, huber_coef)
+    assert_array_almost_equal(huber.intercept_, huber_intercept)
+
+    X, y = make_regression_with_outliers(n_samples=5, n_features=20)
+    X[X < 0.3] = 0.0
+    X_new = np.vstack((X, np.vstack((X[1], X[1], X[3]))))
+    y_new = np.concatenate((y, [y[1]], [y[1]], [y[3]]))
+    huber.fit(X_new, y_new)
+    huber_coef = huber.coef_
+    huber_intercept = huber.intercept_
+    huber.fit(X, y, sample_weight=[1, 3, 1, 2, 1])
+    assert_array_almost_equal(huber.coef_, huber_coef, 3)
+    assert_array_almost_equal(huber.intercept_, huber_intercept, 3)
+
+    # Test sparse implementation with sparse weights.
+    # Checking sparse=non_sparse should be covered in the common tests.
```

Comment:
See https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/utils/estimator_checks.py#L327 . Git blame tells me you wrote it, but I don't blame you for forgetting though ;)